01-30-shield-blocking/actors/enemy/Enemy.gd

```diff
@@ -0,0 +1,26 @@
+extends KinematicBody2D
+class_name Enemy
+
+onready var animated_sprite : AnimatedSprite = $AnimatedSprite
+
+export var move_speed : = 75.0
+export var attack_distance : = 0
+
+var player
+
+
+func _ready() -> void:
+	set_physics_process(false)
+
+
+func _physics_process(delta: float) -> void:
+	var direction = 1 if player.global_position.x > global_position.x else -1
+	move_and_slide(move_speed * Vector2.RIGHT * direction, Vector2.UP)
+	animated_sprite.flip_h = direction
+	if attack_distance != 0 and global_position.distance_to(player.global_position) <= attack_distance:
+		player.take_damage(global_position)
```

Comment:
You shouldn't use a reference to the player to damage them, it's a bad practice to show to people. Use the on_body/area_entered signals on the player and handle the player taking damage in the player scene.

Reply:
What about hit scan enemies? 

If we remove this approach we might as well remove the whole facing/not facing checks. We would end up with a shield that's just a static body which blocks attacks. Unless for some reason said attack hits the hitbox of the player (in case the shield is smaller or what not or actual melee attacks instead of just contact).

Reply:
What does hit scan mean?
You could try to remove the facing checks. Just make the shield's hitbox a vertical rectangle that's tall enough and slightly in front of the character to make sure the enemy can't hit the player. You might need to use areas.  I would add some knockback on both the player and the enemy as well to make the interaction more visual.
You just want to be careful as to where you put the logic. Blocking damage from the attack should happen on the player imo, while enemy knockback and cancelling the enemy's attack should happen on the enemy scene.

Reply:
Hitscan is the approach used in some enemies in doom and Wolfenstein. FPSes in general I believe. Raycast -> hit the player/enemy? -> Damage it. 
I'll add the knockback and remove the `take_damage` call from the enemy.

Reply:
Now that I think about it, the raycast would likely hit the shield 😅 I think I'm overthinking